fraud/registry.go

```diff
@@ -0,0 +1,15 @@
+package fraud
+
+import (
+	"sync"
+)
+
+var unmarshalersLk = sync.RWMutex{}
+var defaultUnmarshalers = map[ProofType]ProofUnmarshaler{}
+
+// Register sets unmarshaler in map by provided ProofType.
+func Register(proofType ProofType, unmarshaler ProofUnmarshaler) {
```

Comment:
Does this have to be exported?

Reply:
Yes, why not? So we can register custom unmarshaler everywhere from code.

Reply:
Yes it has

Reply:
ok.